### DISCUSSION TITLE
Stuck in the npm error while setting up joomla-cms locally, HELP :-(

### DISCUSSION THREAD

**Discussion by ThisIsFaar** [joomla/joomla-cms#32669] [Q&A] [answered]
I was Setting up the Local Environment on my computer.
I clone the Repo on my desktop and then successfully run the composer for PHP dependencies, but when I run `npm ci` I got this error " This is related to npm not being able to find a file."
Please help.

**Comment by dgrammatiko** [comment]
Please provide versions of node and npm. `node -v` and `npm -v`

Fwiw maybe it would be easier to grab the latest node version from their site as you probably have an outdated unsupported version

**Reply by ThisIsFaar** [reply]
node version : v14.16.0
npm version : 6.14.11

**Comment by chmst** [comment, marked as answer]
On which branch of the repo are you? 4.0 dev?

**Reply by richard67** [reply]
Reason for the above question is that we use npm only on the "4.0-dev" branch for Joomla 4, and the error message looks like what you get when trying to run npm on the "staging" or the "3.10-dev" branch.

**Reply by ThisIsFaar** [reply]
Error Screen shot:-
![Screenshot (39)](https://user-images.githubusercontent.com/67395591/111038311-43846e00-844e-11eb-9e36-39cfc12e4661.png)

**Reply by richard67** [reply]
Enter command "git branch -vv" to see on which branch you are, and post here the result.

**Reply by ThisIsFaar** [reply]
result of  this command `git branch -vv`
"staging a7c42a4f8f [origin/staging] link to a field outside of any group (#32646)"

**Reply by brianteeman** [reply]
Looks like you missed a step in the instructions

![image](https://user-images.githubusercontent.com/1296369/111061974-034cdc00-849e-11eb-96a2-79b74308a8b6.png)

**Reply by kevinjoshi46b** [reply]
> Looks like you missed a step in the instructions
> 
> ![image](https://user-images.githubusercontent.com/1296369/111061974-034cdc00-849e-11eb-96a2-79b74308a8b6.png)

Hey please can you provide me with the link for this installation guide to setup a local server.

**Reply by brianteeman** [reply]
https://github.com/joomla/joomla-cms/blob/4.0-dev/README.md

**Reply by kevinjoshi46b** [reply]
Ohhh its present in the "4.0-dev" branch. Thank You.
I actually want to try to work on #30018 so I am trying to setup local environment but I am facing this error while running "composer install"
![image](https://user-images.githubusercontent.com/62877695/111454007-2eebf280-873a-11eb-98e9-a91de9e13c28.png)

**Reply by richard67** [reply]
You have to enable the PHP extensions `gd` and `ldap` in your PHP configuration file `php.ini` as advised in the error message.

**Reply by kevinjoshi46b** [reply]
Thank you @richard67, I figured out how to enable the extensions and the "compose install" command ran successfully :)
![image](https://user-images.githubusercontent.com/62877695/111458642-aec88b80-873f-11eb-9ec5-36ca790272b9.png)

**Reply by kevinjoshi46b** [reply]
And now even the build is successful using the "npm ci" command.
Please can you guide me to the next step (i.e. how do I run the project?)
![image](https://user-images.githubusercontent.com/62877695/111459044-37472c00-8740-11eb-8387-22bac52b3dd6.png)

**Reply by richard67** [reply]
You need a local webserver to use a CMS like Joomla. If you already have that, it would have been better if you had cloned the repository into a sub-folder of your webserver's document root, so you can access it with a web browser.

Either you do that, or you have to copy your stuff all the time from the current folder shown in your screenshot's command line to a sub-folder of your webserver's document root, but that would be silly.

Alternatively you can also configure you webserver in a way so that it uses your folder as a document root.

So or you, if you are at the end where you are now plus your webserver can access it, you navigate to that folder in your webbrowser and then the Joomla 4 installation should start.

**Reply by kevinjoshi46b** [reply]
@richard67 I am sorry if I sound a bit dumb but I have never worked on a CMS before. I have worked directly on developing websites, this is my first time trying to work in developing a CMS. Coming back to your reply I do not have a local webserver. Can you please suggest me a web server that I can use to get joomla up and running locally.

**Reply by richard67** [reply]
On Windows you can use XAMPP https://www.apachefriends.org/index.html .

**Reply by kevinjoshi46b** [reply]
Ohhh I do have xampp installed on my system, what next?

**Reply by kevinjoshi46b** [reply]
So, should I clone the repo in the XAMPP folder?

**Reply by richard67** [reply]
Install it.

Are you one of the potential GSoC 2021 students? If so, all necessary documentation has been linked and pinned in the Glip channel for the GSoC students and mentors.

**Reply by kevinjoshi46b** [reply]
@richard67 Thank You for all your help, I now have everything setup and running. Looking forward to contributing now.

**Comment by richard67** [comment]
@ThisIsFaar Hello. Did one of the above answers solve your problem?

**Reply by ThisIsFaar** [reply]
Hi, @richard67 My problem is solved, thanks :-). pardon me I forgot to mark the answer.

specially thanks to @brianteeman.

final solution: I was running `npm ci` from the wrong branch, So first I switched to 4.0 dev and then everything works like a charm.
